customerio/customerio-ios
# chore: track opened and clicked metrics

https://linear.app/customerio/issue/MBL-331/track-opened-and-clicked-metrics

### Problem
As a marketer, I want to see how effective my inline messages so that I can make better messaging campaigns in the future. I care about how many messages were shown and how many were acted upon (clicked).

### Solution
- When a customer receives an inline message and is displayed on the screen, the SDK is supposed to track `opened` metric for this message.
- When a customer taps on any custom button and takes an action on the button, the SDK is supposed to track `clicked` metric for this message. 

### Demo

https://github.com/user-attachments/assets/694483af-53a5-433d-b443-9a11625af092

### Testing
- Automated test cases in [another PR](https://github.com/customerio/customerio-ios/pull/774)
- For manual testing, send an inline in-app message to the device
- Once it displays, check activity log for `opened` metric
- Tap on action button on inline message 
- Check activity log for `clicked` metric
- Tap multiple times on the same button
- Check activity log, note that there is just once instance of metric tracking. Activity name `inline custom button action` is related to button action but not tracking the inline metric.
